### PR TITLE
Add support for sending email over SES in notification plugin

### LINKED ISSUE #160: Unit test for notification config with SES channel

### PR DESCRIPTION
**Pull request by akbhatta**
### Description
Add support for sending email over SES in notification plugin

[Tests]
Added intergration tests for config operations

Signed-off-by: @akbhatta

### Issues Resolved
#160 #164 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt [hunk 1]
```diff
@@ -19,6 +19,7 @@ import software.amazon.awssdk.regions.Region
  * This class holds the contents of ses destination
  */
 class SesDestination(
+    val accountName: String,
```
Comment (joshuali925) [inline notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt:22]: question, where does accountName come from? i don't recall having this in the UI

**Review by joshuali925 — APPROVED** [review]

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt (hunk 1 above)
Comment (zhongnansu) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt:22]: @joshuali925 Don't we require sender name for SES? (the screenshot is for smtp)?
image

But I do wonder why is accountName being added here, what SES logic is using account/sender name?

Comment (akbhatta) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt:22]: Keeping the model for SMTP and SES somewhat similar

Comment (akbhatta) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt:22]: I believe then sender name to be added to both SMTP and SES?

Comment (joshuali925) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/model/destination/SesDestination.kt:22]: ah i see. yes the UI passes in sender name, which is just called `name` in the `config` object

**Pull request merged**